CRM/Utils/REST.php

```diff
@@ -546,6 +546,14 @@ public static function ajaxJson() {
       );
       CRM_Utils_JSON::output($error);
     }
+    // FIXME: Shouldn't the X-Forwarded-Proto check be part of CRM_Utils_System::isSSL()?
```

Comment:
The standard pattern that I've been able to see for calling redirectToSSL is 

`      // check if ssl is set
      if (!empty($item['is_ssl'])) {
        CRM_Utils_System::redirectToSSL();
      }`

That's not an option here as it relies on menu items.

The bootstrap is needed, otherwise you get undefined function errors.